gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -372,14 +373,18 @@ gst_tensor_filter_transform (GstBaseTransform * trans,
   allocate_in_invoke = gst_tensor_filter_allocate_in_invoke (priv);
 
   /* 1. Set input tensors from inbuf. */
+  /* Internal Logic Error or GST Bug (sinkcap changed!) */
   g_assert (gst_buffer_n_memory (inbuf) == prop->input_meta.num_tensors);
 
   for (i = 0; i < prop->input_meta.num_tensors; i++) {
-    gboolean status;
     in_mem[i] = gst_buffer_peek_memory (inbuf, i);
-    status = gst_memory_map (in_mem[i], &in_info[i], GST_MAP_READ);
-    g_assert (status);
-    /** @todo Do proper error handling (clean up and error return) */
+    if (FALSE == gst_memory_map (in_mem[i], &in_info[i], GST_MAP_READ)) {
+      guint j;
+      for (j = i - 1; j >= 0; j--)
```

Comment:
Q. Would this have problem because of underflow problem when `i` is `0`? (and same for line 408-)

Reply:
Good point. Similar codes will be fixed, too.